staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -171,7 +171,7 @@ func (o *LogsOptions) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().Int64Var(&o.LimitBytes, "limit-bytes", o.LimitBytes, "Maximum bytes of logs to return. Defaults to no limit.")
 	cmd.Flags().BoolVarP(&o.Previous, "previous", "p", o.Previous, "If true, print the logs for the previous instance of the container in a pod if it exists.")
 	cmd.Flags().Int64Var(&o.Tail, "tail", o.Tail, "Lines of recent log file to display. Defaults to -1 with no selector, showing all log lines otherwise 10, if a selector is provided.")
-	cmd.Flags().BoolVar(&o.IgnoreLogErrors, "ignore-errors", o.IgnoreLogErrors, "If watching / following pod logs, allow for any errors that occur to be non-fatal")
+	cmd.Flags().BoolVar(&o.IgnoreLogErrors, "ignore-errors", o.IgnoreLogErrors, false, "If watching / following pod logs, allow for any errors that occur to be non-fatal")
```

Comment:
Thanks for the PR. But this looks behavioral change and I doubt that we can land this. I'd suggest to find another way to fix the issue(I haven't checked the bug).

/hold

Reply:
Ah okay, to summarize, the value of `IgnoreLogErrors` needs to be set to false as default.

Cobra as well as the underlaying module for flags which include `BoolVar()` supports default value assignment, I don't see how this would have led to an error though?

Reply:
@ardaguclu the basic problem is
1. `kubectl logs -f` exits without an issue when the job pod ends
2. `kubectl logs -f` exist without an issue when there's a timeout reading the remote logs

There's AFAIK no way to differentiate the two without doing some retry logic scripting locally, feel free to correct me.